### DISCUSSION TITLE
Getting Started with GitHub Copilot Chat in VSCode

### DISCUSSION THREAD

**Discussion by LadyKerr** [community/community#64517] [Copilot Conversations] [answered]
## Here's how to start using GitHub Copilot Chat in VSCode

GitHub Copilot Chat is an extension that works in your Code Editor or IDE (VS Code or Visual Studio currently) that allows you to have conversations with GitHub Copilot right from your editor.You're able to get code suggestions, build, debug and tests applications with an AI model that understands natural language right from your editor. 

To get started using GitHub Copilot Chat in VSCode, ensure that you have access to the extension by checking your email for access privileges. You get access privileges [ through your organization](https://github.blog/2023-07-20-github-copilot-chat-beta-now-available-for-every-organization/) or by being taken off the[ waitlist for the private beta](https://github.com/github-copilot/chat_waitlist_signup/join) for individuals. You also need to ensure that you have an active GitHub Copilot subscription.

To verify your access, click on your GitHub profile and select "Try Copilot." If you have access to GitHub Copilot, you will see a message at the top of your settings page indicating such and if you don't have access, you will be routed to another page to start your 30-day free trial.

<table>
  <tr>
    <td>
      <figure>
        <img width="1508" alt="copilot-chat-have-access" src="https://user-images.githubusercontent.com/47188731/257664686-4e5fdacf-8e65-4021-81fa-656c52a8148a.png">
         <figcaption>You have access to GitHub Copilot</figcaption>
      </figure>
    </td>
    <td>
      <figure>
        <img width="1400" alt="copilot-chat-access-signup" src="https://user-images.githubusercontent.com/47188731/257663728-98ca93a1-7947-4597-ae71-e0360f062a88.png">
        <figcaption>You do not have access to GitHub Copilot</figcaption>
      </figure>
    </td>
  </tr>
</table>

### Installing GitHub Copilot Chat & asking your first question 

Once you verify those details, follow the steps below to start using GitHub Copilot Chat:

1. Search for "GitHub Copilot Chat" in the vscode extension marketplace, and click the blue install button. After the installation, you'll be prompted to login to your GitHub Account to validate your access to copilot chat.

https://user-images.githubusercontent.com/47188731/257667990-40fa25dc-89ec-44f0-87c5-ae1c44557585.mp4

Be sure to sign in to the GitHub account that has access to your GitHub Copilot subscription and also that has access to Copilot Chat.

2. Once you've installed the extension, you're ready to use it! You'll see a message/chat icon appear on the side of your code editor, click that icon and start asking programming related questions.

You can start by asking a simple question like "How do I set up a new ruby project?" and it will provide you with a suggestion.

https://user-images.githubusercontent.com/47188731/257673009-73f6c260-cc16-4ed2-86d2-4ad28e8e582b.mp4

And that's how you start using copilot chat in VSCode. 

### Slash Commands + GitHub Copilot Chat

There are some really amazing slash commands that you have access to as well that makes it easier to ask for help, create vscode extensions or create unit tests. To start using Slash Commands with GitHub Copilot, type `/` in the input box and you'll see the multiple options come up.

<img width="811" alt="copilot-chat-slash-command" src="https://user-images.githubusercontent.com/47188731/257674956-c0988417-7945-4184-9c24-80d296e8505b.png">

Choose one of the options and then ask a question to get a suggestion. 

### Asking GitHub Copilot Chat questions about your code

To ask Copilot Chat specific questions about the code you're working on, open up the file in your editor, and navigate to the chat extension. 

Let's say for example this is your first time interacting with this particular repository, and you don't quite understand the code, you can ask copilot chat "what does this file mean?" and it will provide a suggested explanation of what the code is doing.

https://user-images.githubusercontent.com/47188731/257677247-6a5f0863-2a54-4874-b16b-0eb2349a2122.mp4 

### Using Code Suggestions from GitHub Copilot Chat 

Whenever you ask copilot chat for code suggestions, you can accept them by copying the suggestion from the chat interface or clicking the "Insert at Cursor" button. You can also insert the suggestion into a new file, or run the code in the terminal if applicable.

https://user-images.githubusercontent.com/47188731/257678263-f046119e-69f6-4e6b-b35f-2323d610bc6b.mp4

I was able to use GitHub Copilot Chat to tell me how to insert two images side by side using markdown ☺️ I copied to provided code suggestion and pasted it here and voila! I had two images side-by-side. 💃🏽

### Opening a Copilot Chat session in your editor

To make your life a little easier, we've made it possible to open up a copilot chat session right in your editor, so you don't have to go back and forth between the chat interface and your files.

To do this, click on the chat icon on the side, and click the three dots at the top and select "Open session in editor" this will move the chat into your editor, similar to how a file would be open.

https://user-images.githubusercontent.com/47188731/257679108-6a06a966-be30-45cf-b856-945a9d378818.mp4

If you have any issues, questions, comments or concerns, feel free to leave them below and I'll try as best as possible to get you an answer! I hope you enjoy using GitHub copilot Chat as much as I do!

Happy Coding!

> [!CAUTION]  
> This content is currently outdated and will not be updated. Last audit date is 2024-03-13.

**Comment by MOHAMEDTITO2030** [comment]
predictor Aviator

**Comment by nielsdt-rabobank** [comment]
I found out that if I am a member of the licensed group of copilot users in two different (enterprise) organisations (prod and test) that both allow Copilot usage, but only one of the two allows copilot chat (the test org), then I will **not** get the chat  icon in Visual Studio Code. 
When using a github id that is a licensed copilot user only of the test org, then I **do** get the icon. 

Any ideas on how I can make it work for the two organisation situation (i.e. I need a group of testers who have "normal" access to copilot through the prod org and temporary access to chat through the test org)?

Appreciate your help.

**Reply by fatemeh644** [reply]
Hi

**Reply by LadyKerr** [reply]
Hey @nielsdt-rabobank thanks for your question! I sent it over to our Solutions Eng who I believe reached out to discuss with you.

**Reply by Danielmatins12345** [reply]
como posso usar este site

**Reply by Shuhrat567** [reply]
thangs

**Comment by RONIE241** [comment]
Access my data account

**Comment by RealJapaneseGreenTea** [comment]
Cooooopilot Hmmmm 🗡️ Yes please.

**Comment by Ling-1109** [comment]
copilot

**Comment by telethrasher** [comment]
@LadyKerr - typically how long will an individual wait for copilot for VSC ?
i just updated to VSCinsiders and waiting for next push update - hopefully that will do it /tt

**Comment by Kimpq** [comment]
Skickat från min iPhone1 sep. 2023 kl. 09:18 skrev Ling-1109 ***@***.***>:﻿
Copilot

**Comment by Kimpq** [comment]
Skickat från min iPhone7 sep. 2023 kl. 01:48 skrev telethrasher ***@***.***>:﻿
@LadyKerr - typically how long will an individual wait for copilot for VSC ?

**Comment by Kimpq** [comment]
Skickat från min iPhone7 sep. 2023 kl. 02:55 skrev Ricard Post ***@***.***>:﻿Skickat från min iPhone1 sep. 2023 kl. 09:18 skrev Ling-1109 ***@***.***>:﻿
Copilot

**Comment by Kimpq** [comment]
Skickat från min iPhone7 sep. 2023 kl. 02:55 skrev Ricard Post ***@***.***>:﻿Skickat från min iPhone7 sep. 2023 kl. 01:48 skrev telethrasher ***@***.***>:﻿
@LadyKerr - typically how long will an individual wait for copilot for VSC ?

**Comment by Kimpq** [comment]
Skickat från min iPhone1 sep. 2023 kl. 09:16 skrev Ling-1109 ***@***.***>:﻿
copilot

**Comment by dylandifilippo** [comment]
Hello,
I'm using Github Copilot Chat on VScode insiders for a while and now it doesn't show up in the sidebar anymore ...
Am I the only one to experience this ?

**Reply by telethrasher** [reply]
had you been using it as an individual or with an 'organization' ?

**Reply by RealJapaneseGreenTea** [reply]
^bump

**Reply by dylandifilippo** [reply, marked as answer]
I solved it. There was an update on VS Code Insiders and it wasn't compatible with Copilot Chat.
I went back to the original VS Code, updated it and then the chat worked.

**Reply by telethrasher** [reply]
which version are you running now that works ?
thx

**Reply by theintrigued** [reply]
I think its visibility was moved at the bottom status bar, that's where it appears for me.

**Reply by dylandifilippo** [reply]
@telethrasher The latest update.

Version: 1.81.1
Commit: 6c3e3dba23e8fadc360aed75ce363ba185c49794
Date: 2023-08-09T22:40:25.698Z
Electron: 22.3.18
ElectronBuildId: 22689846
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Darwin arm64 22.5.0

**Comment by reeveng** [comment]
Can I revert my homepage using github copilot?

**Comment by ferhatkefsiz** [comment]
Thanks!

**Comment by GaryGod3** [comment]
Hi, I've build up a whole bunch of responses from Copilot, some of which I intended going back to as it triggered ideas that I wanted to follow up on.
As yet I can't find a way to search through in the side-panel like you can in other panels, is there something like Ctrl+F available?

**Comment by GAMECHANGE** [comment]
djeva

**Comment by ram-parthiban** [comment]
Why there is no command in the VS code as  `/optimize` separately like the one we have for Visual Studio? @LadyKerr @Akash1134
